### DISCUSSION TITLE
Online upgrading not working

### DISCUSSION THREAD

**Discussion by jmp2000** [aqualinkd/AqualinkD#348] [Q&A] [answered]
Running 2.3.7 - trying to upgrade to 2.4.0.  

When I use the commands shown online for upgrading on the Wiki - I get the following.   

root@aqualinkd:~/software/AqualinkD# cd ~/software/AqualinkD
root@aqualinkd:~/software/AqualinkD# git pull
bash: git: command not found
root@aqualinkd:~/software/AqualinkD# sudo ./release/install.sh
Arch is armhf, Using 32bit AqualinkD
AqualinkD config exists, did not copy new config, you may need to edit existing!     /etc/aqualinkd.conf
AqualinkD defaults exists, did not copy new defaults to /etc/default/aqualinkd
Avahi/mDNS defaults exists, did not copy new defaults to /etc/avahi/services//aqualinkd.service
AqualinkD web config exists, did not copy new config, you may need to edit existing /var/www/aqualinkd//config.js
Starting daemon aqualinkd
root@aqualinkd:~/software/AqualinkD# sudo systemctl start aqualinkd

I then go into Aqmanager to verify what version it is - and it still shows 2.3.7 - so it doesn't look like it performed the upgrade.

I'd prefer to not have to wipe the full RPi and reformat and all that to upgrade between these versions.  Any ideas?   Let me know if you need any other screenshots.

**Comment by sfeakes** [comment]
it looks like you re-installed 2.3.7 over the top of itself.
Look at your 3rd line `bash: git: command not found`

That basically didn't pull the latest version.

**Reply by jmp2000** [reply]
I'm trying to follow the directions on the Wiki:

Upgrading AqualinkD
If you installed AqualinkD using the above you can upgrade AqualinkD to the latest version with these commands.

cd ~/software/AqualinkD
git pull
sudo ./release/install.sh


Any idea what else I should be typing on the command line (once I PuTTy over to the RPi that is running AqualinkD)?

**Reply by sfeakes** [reply]
Did the `git pull` work?   if it failed, and from your previous post looks like you need to install git.  `sudo apt get install git`

BUT, Since you don't have `git`, you probably did the simple install in the beginning.  That has an easier upgrade.
https://github.com/sfeakes/AqualinkD/wiki#SimpleInstall
You may just want to run `~/aqua.sh release`

-- EDIT --
Depending on permissions, you may want to run it as below
```
cd ~
sudo ./aqua.sh release
```

**Reply by jmp2000** [reply]
Apologies... I feel like a total noob here...!   I've tried about every combination.... nothing working for me.   

Any ideas?  Am I in some wrong directory... again apologies... it must be something simple that is clearly eluding me!

root@aqualinkd:~# ~/aqua.sh release
bash: /root/aqua.sh: No such file or directory
root@aqualinkd:~# cd software
root@aqualinkd:~/software# dir
AqualinkD  AqualinkD-2.3.7
root@aqualinkd:~/software# ~/aqua.sh release
bash: /root/aqua.sh: No such file or directory
root@aqualinkd:~/software# cd ~
root@aqualinkd:~# dir
software
root@aqualinkd:~# sudo ./aqua.sh release
sudo: ./aqua.sh: command not found
root@aqualinkd:~# sudo su
root@aqualinkd:~# cd ~
root@aqualinkd:~# sudo ./aqua.sh release
sudo: ./aqua.sh: command not found
root@aqualinkd:~# ^C
root@aqualinkd:~# ~/aqua.sh release
bash: /root/aqua.sh: No such file or directory
root@aqualinkd:~#

**Reply by jmp2000** [reply]
I did install git - "sudo apt-get install git", after i did an update (it gave me errors before then)..  sudo apt-get update.

The Raspbian update worked, and then installation of Git worked.   But then i tried about everything else.... still no go.   Checked on Aqmanager - still running 2.3.7.    Oy =)

root@aqualinkd:~/software/AqualinkD# ~/aqua.sh release
bash: /root/aqua.sh: No such file or directory
root@aqualinkd:~/software/AqualinkD# git pull
fatal: not a git repository (or any of the parent directories): .git
root@aqualinkd:~/software/AqualinkD# sudo ./release/install.sh
Arch is armhf, Using 32bit AqualinkD
AqualinkD config exists, did not copy new config, you may need to edit existing! /etc/aqualinkd.                 conf
AqualinkD defaults exists, did not copy new defaults to /etc/default/aqualinkd
Avahi/mDNS defaults exists, did not copy new defaults to /etc/avahi/services//aqualinkd.service
AqualinkD web config exists, did not copy new config, you may need to edit existing /var/www/aqu                 alinkd//config.js
Starting daemon aqualinkd
root@aqualinkd:~/software/AqualinkD# git pull sudo ./release/install.sh
fatal: not a git repository (or any of the parent directories): .git
root@aqualinkd:~/software/AqualinkD# sudo systemctl start aqualinkd
root@aqualinkd:~/software/AqualinkD# sudo systemctl stop aqualinkd
root@aqualinkd:~/software/AqualinkD# cd ~
root@aqualinkd:~# sudo ./aqua.sh release
sudo: ./aqua.sh: command not found
root@aqualinkd:~# cd ~/software/AqualinkD
root@aqualinkd:~/software/AqualinkD# sudo ./aqua.sh release
sudo: ./aqua.sh: command not found
root@aqualinkd:~/software/AqualinkD# aqua.sh release
bash: aqua.sh: command not found
root@aqualinkd:~/software/AqualinkD# git pull
fatal: not a git repository (or any of the parent directories): .git
root@aqualinkd:~/software/AqualinkD#

**Reply by sfeakes** [reply, marked as answer]
The below is in reply too https://github.com/sfeakes/AqualinkD/discussions/348#discussioncomment-10653817

It looks like you are getting confused with standard user and root user.
~ means home directly, when you are standard user that's different to when you are root.  When root ~ means /root, then user ~ means /home/user.
I think you probably have aqua.sh in a users home directory, but every command you try you are root, so it's trying to find that in root home directly.

I'm not sure what your regular user name is, so I can't give you the commands to run. BUT, best bet.  Start a new ssh session, that will log you in as standard user in home directory.
run.
``` ls -a aqua.sh```
If you get a file then run
```sudo ./aqua.sh release```
If you get no file found then run
```curl -s https://raw.githubusercontent.com/sfeakes/AqualinkD/master/extras/aqua.sh -o aqua.sh && chmod 755 aqua.sh && ./aqua.sh release```

That should fix things up for you.

**Reply by sfeakes** [reply]
> I did install git - "sudo apt-get install git", after i did an update (it gave me errors before then).. sudo apt-get update.
> 
> The Raspbian update worked, and then installation of Git worked. But then i tried about everything else.... still no go. Checked on Aqmanager - still running 2.3.7. Oy =)
> 
> root@aqualinkd:~/software/AqualinkD# git pull fatal: not a git repository (or any of the parent directories): .git 

That is telling you are not in a git repo.  This is because you must have done the "simple / trust everything install" initially.

If you want to move to the long / manual install, then you will need to follow all the git commands below.
https://github.com/sfeakes/AqualinkD/wiki#Install

BUT, you are probably better off using the simple install, so forget the git commands.

**Reply by jmp2000** [reply]
Ha!  One always feels dumb once the answer is pointed out at your feet!   Thank you for your patience... like every day, I learned something new there!    Thank you so much!   (Version 2.4.0 works like a charm - like the multiple temp indications - they run great!!)

**Reply by sfeakes** [reply]
Glad you got it working/upgraded.  Also good that the multiple temp are working.
